src/nvim/CMakeLists.txt

```diff
@@ -639,17 +639,26 @@ if(CLANG_ASAN_UBSAN)
       set(SANITIZE_RECOVER -fno-sanitize-recover)      # Clang 3.5-
     endif()
   endif()
-  set_property(TARGET nvim APPEND PROPERTY COMPILE_OPTIONS ${SANITIZE_RECOVER} -fno-omit-frame-pointer -fno-optimize-sibling-calls -fsanitize=address -fsanitize=undefined -fsanitize-blacklist=${PROJECT_SOURCE_DIR}/src/.asan-blacklist)
-  set_property(TARGET nvim APPEND_STRING PROPERTY LINK_FLAGS "-fsanitize=address -fsanitize=undefined ")
+  target_compile_options(nvim PRIVATE
+    ${SANITIZE_RECOVER}
+    -fno-omit-frame-pointer
+    -fno-optimize-sibling-calls
+    -fsanitize=address
+    -fsanitize=undefined
+    -fsanitize-blacklist=${PROJECT_SOURCE_DIR}/src/.asan-blacklist)
```

Comment:
I would suggest to use `${nvim_SOURCE_DIR}`. You never know if the project is included by a another one and this would point to the wrong one :-)

Reply:
Hmmm, wouldn't the same dilemma exist for the parent project? They wouldn't be able to use `PROJECT_SOURCE_DIR` in case another project includes *their* project :)

That being said: I don't mind this, so I'll include it 👍

Reply:
Actually in hindsight, the mixing of lowercase and uppercase is particularly fugly, even for cmake standards. Is there an external project this currently breaks?

Reply:
I have no idea. I learned it the hard way through bug reports for two of my projects (cmocka, libssh).

Reply:
Honestly it's too ugly lol. I'm willing to risk it. You can tell me "I told you so" if this ends up biting me in the ass later on :)

Reply:
Also this is a relatively unknown feature, so it's gonna be annoying for new contributors to figure out where the variable came from.